chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1846,3 +1800,124 @@ async fn filter_call_triggers_from_unsuccessful_transactions(
     }
     Ok(block)
 }
+
+/// Deprecated. Wraps the [`fetch_transaction_receipts_in_batch`] in a retry loop.
+async fn fetch_transaction_receipts_in_batch_with_retry(
+    web3: Arc<Web3<Transport>>,
+    hashes: Vec<H256>,
+    block_hash: H256,
+    logger: Logger,
+) -> Result<Vec<TransactionReceipt>, IngestorError> {
+    retry("batch eth_getTransactionReceipt RPC call", &logger)
+        .limit(*REQUEST_RETRIES)
+        .no_logging()
+        .timeout_secs(*JSON_RPC_TIMEOUT)
+        .run(move || {
+            let web3 = web3.cheap_clone();
+            let hashes = hashes.clone();
+            let logger = logger.cheap_clone();
+            fetch_transaction_receipts_in_batch(web3, hashes, block_hash, logger).boxed()
+        })
```

Comment:
Those clones are needed because those values are sent (possibly multiple times) to the `fetch_transaction_receipts_in_batch`, at line 1819.